examples/specialty_plots/swotplot.py

```diff
@@ -0,0 +1,180 @@
+        Data structure passed to populate the quadrant.
+        Format 'key':Tuple(float,float)
+    ax : plt.Axes
+        Matplotlib current axes
+    q_color : str
+        Quadrant color.
+    q_x : Tuple[float]
+        Quadrant X relative coordinates. Format: Tuple(float,float,float,float)
+    q_y : Tuple[float]
+        Quadrant Y relative coordinates. Format: Tuple(float,float,float,float)
+    q_title_pos : Tuple[float,float]
+        Plot title relative position
+    q_title_props : dict, optional
+        Title box style properties. The default is dict(boxstyle='round',
+        facecolor="white", alpha=1.0).
+    q_point_props : dict, optional
+        Point box style properties. The default is dict(boxstyle='round',
+        facecolor="white", alpha=1.0).
+
+    Returns
+    -------
+    None
+
+    """
+    # draw filled rectangle
+    ax.fill(q_x, q_y, color=q_color)
+
+    # draw quadrant title
+    ax.text(x=q_title_pos[0], y=q_title_pos[1],
+            s=str.upper(list(q_data.keys())[0]), fontsize=12, weight='bold',
+            verticalalignment='center', horizontalalignment='center',
+            bbox=q_title_props)
+
+    # fill quadrant with points
+    for k, v in q_data[list(q_data.keys())[0]].items():
+        ax.text(x=v[0], y=v[1], s=k, fontsize=12, weight='bold',
+                verticalalignment='center', horizontalalignment='center',
+                bbox=q_point_props)
+        _log.debug(f"k: {k}, v: {v}")
+
+    return None
+
+
+def swotplot(p_data: dict, ax: plt.Axes,
+             s_color: str = "forestgreen", w_color: str = "gold",
+             o_color: str = "skyblue", t_color: str = "firebrick",
+             left_margin: float = 0.05, right_margin: float = 0.05,
+             top_margin: float = 0.05, bottom_margin: float = 0.05,
+             ) -> None:
+    """
+    Draw SWOT plot
+
+    Parameters
+    ----------
+    p_data : dict
+        Data structure passed to populate the plot.
+    ax : matplotlib.pyplot.Axes
+        axes in which to draw the plot
+    s_color : float, optional
+        Strength quadrant color.
+    w_color : float, optional
+        Weakness quadrant color.
+    o_color : float, optional
+        Opportunity quadrant color.
+    t_color : float, optional
+        Threat quadrant color.
+    left_margin : float, optional
+        Left spacing from frame border. The default is 0.05.
+    right_margin : float, optional
+        Right spacing from frame border. The default is 0.05.
+    top_margin : float, optional
+        Top spacing from frame border. The default is 0.05.
+    bottom_margin : float, optional
+        Bottom spacing from frame border. The default is 0.05.
+
+
+    Returns
+    -------
+    None
+
+    """
+    # format axis
+    ax.set_xlim(0, 1.0)
+    ax.set_ylim(0, 1.0)
+    ax.axis('off')
+
+    # draw s quadrant
+    draw_quadrant(q_data=dict(filter(lambda i: i[0] in list(p_data.keys())[0],
+                                     p_data.items())),
```

Comment:
If I've understood this correctly, are you expecting the input to be a dictionary with 4 items, and the order refers to each quadrant? If so, I'm not sure that you should use a dictionary for that. Maybe a list of 4 (name, quadrant data) tuples or 4 arguments of the same would be better.

Reply:
See comment below